Response decompression: rework batch test to WireMock

vlsi opened the pull request:
Note: this builds on top of https://github.com/apache/jmeter/pull/496

## Motivation and Context

ResponseDecompression test fails too often:
https://travis-ci.org/apache/jmeter/jobs/587884339#L1458
https://travis-ci.org/apache/jmeter/jobs/587201685#L1457
https://travis-ci.org/apache/jmeter/jobs/585431878#L1437
https://travis-ci.org/apache/jmeter/jobs/585146534#L1452
https://travis-ci.org/apache/jmeter/jobs/584778862#L1450
https://travis-ci.org/apache/jmeter/jobs/582277722#L1204

So I refactored it to use WireMock

Unfortunately WireMock does not support `deflate` and `br` compressions, so those are still send requests to the Internet.

vlsi committed:
Rework ResponseDecompression test to use WireMock for gzip/nocompression

This makes the test more stable

ham1 commented on src/dist-check/src/test/java/org/apache/jmeter/junit/JMeterTest.java:
```diff
@@ -113,6 +115,20 @@ public JMeterTest(String testName, JMeterGUIComponent gc) {
+        StringBuilder sb = new StringBuilder();
+        sb.append(getName());
+        if (guiItem instanceof TestBeanGUI) {
+            sb.append(" ").append(guiItem.toString());
```
Could this repeated `.append(" ")` be added before the `if`s?

vlsi replied:
This commit is not part of this PR. This PR builds on top of https://github.com/apache/jmeter/pull/496, so please comment JUnit5 migration there.

ham1 reviewed (commented):
:+1:

pmouawad commented:
Hi @vlsi ,
Great idea.
But I think failure mainly comes from Deflate no ?
1/ Why did you remove HR1-DEFLATE sampler ? is it not Deflate ?
2/ My understanding is that it will not fix root cause no ?

But I am ok with PR idea provided HR1-DEFLATE is put back again unless there is a reason (I didn't deeply look still).
Thanks

vlsi commented:
>But I think failure mainly comes from Deflate no ?

Yes and no. Mainly the failure comes from using the unreliable internet for JMeter testing.
So I would treat "removal / replacement of the URLs" to be workaround rather than a solution.

>1/ Why did you remove HR1-DEFLATE sampler ? is it not Deflate ?

It causes failures. Really. Every time the test fails it was caused by HR1-DEFLATE.

>2/ My understanding is that it will not fix root cause no ?

Removal of the ill site would heal the test.

However I don't think testing deflate is important, since, well, is it really used in practice?

pmouawad commented:
> > But I think failure mainly comes from Deflate no ?
> 
> Yes and no. Mainly the failure comes from using the unreliable internet for JMeter testing.

Is it network or DDOS protection on server side ?

> So I would treat "removal / replacement of the URLs" to be workaround rather than a solution.
> 
> > 1/ Why did you remove HR1-DEFLATE sampler ? is it not Deflate ?

The idea was to have more than 1 site in test.
> 
> It causes failures. Really. Every time the test failed it was caused by HR1-DEFLATE.
> 
> > 2/ My understanding is that it will not fix root cause no ?
> 
> Removal of the ill site would heal the test.
> 
> However I don't think testing deflate is important, since, well, is it really used in practice?

I think it is. We had bug report, and within our consulting work at UbikLoadPack we frequently see it.

vlsi commented:
>Is it network or DDOS protection on server side ?

probably

I think brotli and deflate could be reworked to use Docker-based nginx server.

pmouawad commented:
Could we use this :
https://www.testcontainers.org/

vlsi committed:
Move to JUnit5 for JUnit test execution

Note: JUnit3 and JUnit4 tests can still be executed, however it makes sense to use JUnit5 for new tests

Signed-off-by: Vladimir Sitnikov <sitnikov.vladimir@gmail.com>
